spec/tbd_tests_spec.rb

```diff
@@ -4708,6 +4601,193 @@
+    expect(TBD.status).to eq(0)
+
+    # Order of arguments matter.
+    expect(fits?(wall.vertices, door1.vertices)).to be(false)
+    expect(overlaps?(wall.vertices, door1.vertices)).to be(true)
+    expect(TBD.status).to eq(0)
+
+    # Another 1m x 2m corner door, yet entirely beyond the wall surface.
+    vec = OpenStudio::Point3dVector.new
+    vec << OpenStudio::Point3d.new( 16,  0,  2)
+    vec << OpenStudio::Point3d.new( 16,  0,  0)
+    vec << OpenStudio::Point3d.new( 17,  0,  0)
+    vec << OpenStudio::Point3d.new( 17,  0,  2)
+    door2 = OpenStudio::Model::SubSurface.new(vec, model)
+    ft_door2 = flatZ( (ft * door2.vertices).reverse )
+    union = OpenStudio::join(ft_wall, ft_door2, TOL2)
+    expect(union.empty?).to be(true)
+
+    # Door2 fits?, overlaps?
+    expect(fits?(door2.vertices, wall.vertices)).to be(false)
+    expect(overlaps?(door2.vertices, wall.vertices)).to be(false)
+    expect(TBD.status).to eq(0)
+
+    # # Order of arguments doesn't matter.
+    expect(fits?(wall.vertices, door2.vertices)).to be(false)
+    expect(overlaps?(wall.vertices, door2.vertices)).to be(false)
+    expect(TBD.status).to eq(0)
+
+    # Top-right corner 2m x 2m window, overlapping top-right corner of wall.
+    vec = OpenStudio::Point3dVector.new
+    vec << OpenStudio::Point3d.new(  9,  0, 11)
+    vec << OpenStudio::Point3d.new(  9,  0,  9)
+    vec << OpenStudio::Point3d.new( 11,  0,  9)
+    vec << OpenStudio::Point3d.new( 11,  0, 11)
+    window = OpenStudio::Model::SubSurface.new(vec, model)
+    ft_window = flatZ( (ft * window.vertices).reverse )
+    union = OpenStudio::join(ft_wall, ft_window, TOL2)
+    expect(union.empty?).to be(false)
+    union = union.get
+    area = OpenStudio::getArea(union)
+    expect(area.empty?).to be(false)
+    area = area.get
+    expect(area).to be_within(0.01).of(103)
+
+    # Window fits?, overlaps?
+    expect(fits?(window.vertices, wall.vertices)).to be(false)
+    expect(overlaps?(window.vertices, wall.vertices)).to be(true)
+    expect(TBD.status).to eq(0)
+
+    expect(fits?(wall.vertices, window.vertices)).to be(false)
+    expect(overlaps?(wall.vertices, window.vertices)).to be(true)
+    expect(TBD.status).to eq(0)
+
+    # A glazed surface, entirely encompassing the wall.
+    vec = OpenStudio::Point3dVector.new
+    vec << OpenStudio::Point3d.new(  0,  0, 10)
+    vec << OpenStudio::Point3d.new(  0,  0,  0)
+    vec << OpenStudio::Point3d.new( 10,  0,  0)
+    vec << OpenStudio::Point3d.new( 10,  0, 10)
+    glazing = OpenStudio::Model::SubSurface.new(vec, model)
+
+    # Glazing fits?, overlaps?
+    expect(fits?(glazing.vertices, wall.vertices)).to be(true)
+    expect(overlaps?(glazing.vertices, wall.vertices)).to be(true)
+    expect(TBD.status).to eq(0)
+
+    expect(fits?(wall.vertices, glazing.vertices)).to be(true)
+    expect(overlaps?(wall.vertices, glazing.vertices)).to be(true)
+    expect(TBD.status).to eq(0)
+  end
+
+  it "can calculate offset for (frame & dividers)" do
```

Comment:
I added this test just to show how to use the `OpenStudio::buffer` method.  If you want you could update your `offset` method to use `buffer` or not, whichever you prefer.  For a non-ninety degree angle, the `buffer` method accounts for the extra height/gap/whatever it's called that is required to join the two offsets at a single point.